### ISSUE TITLE
Merge / concatenate JavaScript files

### ISSUE THREAD

**Issue by DavidSvetlik** [webpack/webpack#4425]
**Do you want to request a *feature* or report a *bug*?**

Request a feature.

**What is the current behavior?**

Webpack 2 apparently doesn't support this.

**If the current behavior is a bug, please provide the steps to reproduce.**

X

**What is the expected behavior?**

X

**If this is a feature request, what is motivation or use case for changing the behavior?**

Hello,

I had set-up Webpack 2 project with entry points for both main and vendor bundles, however I was wondering if I could use Webpack to build another bundle which would be used both inside and outside of an application, so it shouldn't be packed within Webpack webpackJsonp wrapper in order to stay independent and expose everything it wants.

For an idea, it would contain es5/6 shim&sham (node modukes) plus some custom pollyfills, like this:

```
import 'es5-shim';
import 'es5-shim/es5-sham';
import 'es6-shim';
import 'vendor/polyfills.js';
```

Basically just concatenate and minify a few files.

I was searching through the internet and documentation for an answer without any success.

I know Webpack has a specific use case, but it seems kind of dumb to install something else just for this specific task (or worse, do it manually), especially when I already use Webpack.

Thank you.

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**

X

**Comment by timse** [comment]
do you mean library?

**Comment by DavidSvetlik** [comment]
@timse Sorry, don't understand your question. *Where* am I supposed to mean library?

**Comment by probablyup** [comment]
@DavidSvetlik Do you mean running webpack build a second time with `commonjs ` as the `output.libraryTarget`? https://webpack.js.org/configuration/output/#output-librarytarget

That would allow the bundle to be consumed elsewhere without the JSONP stuff.

**Comment by timse** [comment]
ill close this for now, as it seems to not be active anymore. If you feel like your question was not answered feel free to reopen.

**Comment by DavidSvetlik** [comment]
@timse Not sure to be honest. I don't want to create a library as @probablyup suggests. I want to bundle few existing libraries into one file within same project (config).

After reading the documentation it seems though that Webpack 2 cannot do that (or maybe can, I wasn't able to find any working example), which is a shame since it's quite trivial task (just concatenate and minify) and yet I'll have to search for another tool for it. That's why I marked this issue as a feature request in the first place.

Thanks though for all the help.

**Comment by ThatTomPerson** [comment]
would just using an array for an entry not work?
eg
```
{
  entry: {
    app: './app.js',
    concated: ['es5-shim', 'es5-shim/es5-sham', 'es6-shim', './vendor/polyfills.js']
  },
  output: {
    filename: '[name].js'
  }
}
```
this would output `app.js` and `concated.js`

**Comment by DavidSvetlik** [comment]
@ThatTomPerson Yeah, but that would still create webpack jsonp package. I need it to be separate from the rest so I can embed it both inside and outside of an application.

**Comment by ThatTomPerson** [comment]
So you could use the multicompiler then
```
[
  {
    entry: './app.js',
    output: {
      filename: 'app.js'
    }
  },
  {
    entry: ['es5-shim', 'es5-shim/es5-sham', 'es6-shim', './vendor/polyfills.js'],
    output: output: {
      filename: 'concated.js'
    }
]
```

**Comment by DavidSvetlik** [comment]
@ThatTomPerson That's it! Thank you.

**Comment by justinwhall** [comment]
I too, am looking to do what @DavidSvetlik is. @ThatTomPerson, your suggestion makes sense to me conceptually. Pass entires as an array basically?

Little crossed up on syntax as my config looks a bit different. `publicJS` is my vendor js that I would like to simply concatenate. 

`
const publicJS = [
  './js/bootstrap.js',
  './js/fancybox/source/jquery.fancybox.pack.js', 
  './js/flexslider/jquery.flexslider-min.js', 
  './js/bxslider/jquery.bxslider.min.js',
  './js/jquery.validate/jquery.validate.min.js', 
  './js/jquery.validate/additional-methods.min.js', 
  './js/jquery.maskedinput.min.js', 
  './js/jquery.formatDateTime.js',
  './js/gpw-public.js'
];

const config = {
  devtool: 'eval-source-map',
  context: path.resolve(__dirname, ''),
  devServer: {
      proxy: {
        '**': {
          target: 'http://www.gdwv2.com',
          secure: false,
          changeOrigin: true
        }
      }
  },
  
  entry: {
    './js/theme-selector/gpw-theme-selector.min': './js/theme-selector/gpw-theme-selector.js',
    './pub/public.bundle': publicJS,
  },
  output: {
    path: path.resolve(__dirname, ''),
    // publicPath: '',
    filename: '[name].js'
  },
  module: {
    // noParse: /jquery|lodash/,
    rules: [{
      test: /\.(png|jpg)$/,
      include: path.resolve(__dirname, 'style'),
      use: [{
        loader: 'url-loader',
        options: { limit: 10000 } // Convert images < 10k to base64 strings
      }]
      }, 
      {
        // 
        test: /\.scss$/,
        include: path.resolve(__dirname, 'style'),
        loader: extractCSS.extract(['css-loader','sass-loader'])
      }, 
      {
        // Theme selector
        test: /\.js$/,
        include: path.resolve(__dirname, 'js/theme-selector'),
        use: [{
          loader: 'babel-loader',
          options: { presets: ['es2015', 'react'] }
        }]
      },
      {
        test: /\.js$/,
        include: path.resolve(__dirname, 'js'),
        use: [{
          loader: 'babel-loader',
          options: { presets: ['es2015', 'react'] }
        }]
      }
    ]
  },
  plugins: [
    new webpack.NamedModulesPlugin(),
    extractCommons,
    extractCSS
  ]
}
`

**Comment by liqiang372** [comment]
I think it is more reasonable for webpack to provide an option to decide if it's going to parse the files or simply concatenate an array of files instead of using another compiler.

**Comment by PetersonLian** [comment]
@liqiang372 I agreed. I am now encountering a situation where i just need to concatenate an array of files without compiling or any other operations. Now i am stuck, any solution (just using webpack only.no gulp or other tools)

**Comment by liqiang372** [comment]
@PetersonLian Yeah, even if you don't do anything, webpack will automatically wrap them in jsonp block. The best way is to concat files with your own scripts. check npm for concat

**Comment by JCarlosR** [comment]
I have 2 files with no module dependencies.
I just want to minify each file, and omit the webpack bootstrap code and jsonp.
Is it not possible to do with webpack? :(

**Comment by hristokostov** [comment]
Maybe this plugin - [webpack-concat-plugin](https://github.com/hxlniada/webpack-concat-plugin). But it's limited and buggy. Currently trying to workaround some problems and integrate into my workflow.

**Comment by iapYang** [comment]
Any progress here?
